tasks/smoke-test/util.ts

```diff
@@ -0,0 +1,10 @@
+import isPortReachable from 'is-port-reachable'
+
+export function waitForServer(port, interval) {
+  return new Promise((resolve) => {
+    setInterval(async () => {
+      const isServerUp = await isPortReachable(port, { host: 'localhost' })
+      isServerUp && resolve(true)
+    }, interval)
+  })
```

Comment:
`isPortReachable` takes a [`timeout`](https://github.com/sindresorhus/is-port-reachable#timeout). Maybe we could use that to the same effect?
```suggestion
return isPortReachable(port, { host: 'localhost', timeout })
```

Reply:
Hmmm let me check.... maybe it doens't in v3 (the non esm version), but great suggestion

Reply:
ah right, forgot we're not on the latest

Reply:
So timeout _is_ available in v3.x too - but it doesn't seem to do what we were hoping. In the case of the dev server it just launches the server and waits, doesn't ever seem to think the server is up.